src/main/java/io/vertx/core/http/impl/HttpServerImpl.java

```diff
@@ -1019,5 +1065,13 @@ private void http1(ChannelHandlerContext ctx, ByteBuf buf) {
       ctx.fireChannelRead(buf);
       ctx.pipeline().remove(this);
     }
+
+    @Override
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+      Channel channel = ctx.channel();
+      HandlerHolder<HttpHandler> reqHandler = reqHandlerManager.chooseHandler(channel.eventLoop());
+      reqHandler.context.executeFromIO(channel::close);
```

Comment:
Done

2016-09-14 18:25 GMT+02:00 Julien Viet <notifications@github.com>:

> In src/main/java/io/vertx/core/http/impl/HttpServerImpl.java
> <https://github.com/eclipse/vert.x/pull/1620#discussion_r78783851>:
>
> > @@ -1019,5 +1065,13 @@ private void http1(ChannelHandlerContext ctx, ByteBuf buf) {
> >        ctx.fireChannelRead(buf);
> >        ctx.pipeline().remove(this);
> >      }
> > +
> > +    @Override
> > +    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
> > +      Channel channel = ctx.channel();
> > +      HandlerHolder<HttpHandler> reqHandler = reqHandlerManager.chooseHandler(channel.eventLoop());
> > +      reqHandler.context.executeFromIO(channel::close);
>
> actually channel.close should be called normally but it's the callback to
> connectionExceptionHandler that should be executed with executeFromIO.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse/vert.x/pull/1620/files/8c4915a0574b4970668c125f10b611dfad7fb4ad#r78783851>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABbltqGI1qNoqpvfTe2QO406X0hqfYuaks5qqCANgaJpZM4J44vk>
> .
>